components/arm.py

```diff
@@ -161,7 +233,11 @@ def set_length(self, value: float) -> None:
         """Sets a goal length to go to in meters"""
         self.goal_extension = clamp(value, self.MIN_EXTENSION, self.MAX_EXTENSION)
 
-    DEFAULT_ALLOWABLE_ANGLE_ERROR = math.radians(5)
+    def set_setpoint(self, value: "Setpoint") -> None:
```

Comment:
nit: Can you move the Setpoint class higher up to avoid forward references?

Reply:
the setpoints class needs the arm for its MIN/MAX_EXTENSION/ANGLE and i thought the Setpoint and Setpoints classes would be better next to each other

Reply:
Would it make sense to pull those constants out of the Arm class?